.github/workflows/qa_build_and_deploy.yml

```diff
@@ -60,7 +61,36 @@ jobs:
 
       - name: Prepare .env file for the deployment
         id: variables_population
-        run: echo "VIRTUAL_HOST=$VIRTUAL_HOST \nINTEGRATED_TURN_IP=$INTEGRATED_TURN_IP \nTAG=$TAG" > .env
+        env: 
+          DB_USERNAME: ${{ secrets.DB_USERNAME }}
+          DB_PASSWORD: ${{ secrets.DB_PASSWORD }}
+          GF_SECURITY_ADMIN_PASSWORD: ${{ secrets.GF_SECURITY_ADMIN_PASSWORD }}
+          GF_SECURITY_ADMIN_USER: ${{ secrets.GF_SECURITY_ADMIN_USER  }}
+        run: |
+          echo "VIRTUAL_HOST=$VIRTUAL_HOST 
+          INTEGRATED_TURN_IP=$INTEGRATED_TURN_IP 
+          TAG=$TAG 
+          DB_USERNAME=$DB_USERNAME 
+          DB_PASSWORD=$DB_PASSWORD
+          GF_SECURITY_ADMIN_PASSWORD=$GF_SECURITY_ADMIN_PASSWORD
+          GF_SECURITY_ADMIN_USER=$GF_SECURITY_ADMIN_USER" > .env
+
+      # commands executed in action below are packed into .sh script, to avoid workflow crashes when some containers or volumes don't exist
+      - name: Remove Grafana and Videoroom containers with volumes
+        uses: JimCronqvist/action-ssh@master
+        with:
+          hosts: ${{ secrets.QA_HOST }}
+          privateKey: ${{ secrets.SSH_PRIV_KEY }}
+          command: |
+            echo "docker kill ${{ env.PROJECT }}_videoroom_1
+            docker kill ${{ env.PROJECT }}_grafana_1
+            docker rm ${{ env.PROJECT }}_videoroom_1
+            docker rm ${{ env.PROJECT }}_grafana_1
+            docker volume rm ${{ env.PROJECT }}_grafana-provisioning
+            echo ok" > docker_kill.sh
+            chmod +x docker_kill.sh
+            ./docker_kill.sh
+            rm docker_kill.sh
```

Comment:
It doesn't seem the right way. My quick manual test using ssh and based on the code of `JimCronqvist/action-ssh` suggested that adding `echo ""` as a last step in commands should solve the problem as only the last command's return value is also returned by `ssh`.

```suggestion
      - name: Remove Grafana and Videoroom containers with volumes
        uses: JimCronqvist/action-ssh@master
        with:
          hosts: ${{ secrets.QA_HOST }}
          privateKey: ${{ secrets.SSH_PRIV_KEY }}
          command: |
            docker rm -f ${{ env.PROJECT }}_videoroom_1
            docker rm -f -v ${{ env.PROJECT }}_grafana_1
            docker volume rm ${{ env.PROJECT }}_grafana-provisioning
            echo ""
```

If that doesn't help and you'll need every command to succeed, use `|| true`

```suggestion
      - name: Remove Grafana and Videoroom containers with volumes
        uses: JimCronqvist/action-ssh@master
        with:
          hosts: ${{ secrets.QA_HOST }}
          privateKey: ${{ secrets.SSH_PRIV_KEY }}
          command: |
            docker rm -f ${{ env.PROJECT }}_videoroom_1 || true
            docker rm -f ${{ env.PROJECT }}_grafana_1 || true
            docker volume rm ${{ env.PROJECT }}_grafana-provisioning || true
```

Reply:
It seems, that calling `docker rm -f` or `docker volume rm -f` returns 0, even if specific image or volume does not exists